pdi-execution-engine/api/src/main/java/org/pentaho/di/engine/api/IOperation.java

```diff
@@ -11,9 +12,9 @@
  * which is associated with the behavior specific to
  * an IEngine.
  */
-public interface IOperation extends Serializable {
+public interface IOperation extends IReportingEventSource {
 
-  String getId();
+  @Override String getId();
```

Comment:
@pentaho-nbaker - remove getId()?  Don't we need some way of identifying the operation?